### DISCUSSION TITLE
Is it expected that UDP:port changes for SRP operations?

### DISCUSSION THREAD

**Discussion by AlanLCollins** [openthread/openthread#8761] [Q&A] [answered]
I have seen runs which my device uses udp:ports 53536 and 53538, instead of the IANA 53535 port. 
Is this normal operation? 

please refer to sniffer log attached.
network key: 2fd97137be877ea903707b9f721b08bb 
[Thread_sniffer.pcapng.zip](https://github.com/openthread/openthread/files/10735661/Thread_sniffer.pcapng.zip)

![image](https://user-images.githubusercontent.com/93798708/218824700-b37c3e7b-686b-4dc4-872e-fa8282e1382c.png)

**Comment by jwhui** [comment, marked as answer]
When publishing the SRP server's unicast address, the UDP port is used to trigger devices to re-register with the SRP server. For example, because the SRP server does not maintain service registrations in non-volatile storage, the SRP server will change the UDP port when it reboots.

The range of UDP ports is configured as below:

https://github.com/openthread/openthread/blob/d28793c104ab710f692a160377a5753373ae4b69/src/core/config/srp_server.h#L63-L81
